src/GovUk.Education.ExploreEducationStatistics.Notifier/Functions/SubscriptionManager.cs

```diff
@@ -10,155 +11,141 @@
-            _tokenService = tokenService ?? throw new ArgumentNullException(nameof(tokenService));
-            _emailService = emailService ?? throw new ArgumentNullException(nameof(emailService));
-            _storageTableService = storageTableService ?? throw new ArgumentNullException(nameof(storageTableService));
-            _notificationClientProvider = notificationClientProvider ??
-                                          throw new ArgumentNullException(nameof(notificationClientProvider));
-        }
-
-        [Function("PublicationSubscribe")]
+        private readonly AppSettingOptions _appSettingOptions = appSettingOptions.Value;
+        private readonly GovUkNotifyOptions.EmailTemplateOptions _emailTemplateOptions = govUkNotifyOptions.Value.EmailTemplates;
+        private readonly ITokenService _tokenService = tokenService ?? throw new ArgumentNullException(nameof(tokenService));
+        private readonly IEmailService _emailService = emailService ?? throw new ArgumentNullException(nameof(emailService));
+        private readonly IStorageTableService _storageTableService = storageTableService ?? throw new ArgumentNullException(nameof(storageTableService));
+        private readonly INotificationClientProvider _notificationClientProvider = notificationClientProvider ??
+                                                                                   throw new ArgumentNullException(nameof(notificationClientProvider));
+
+        [Function("RequestPendingSubscription")]
         // ReSharper disable once UnusedMember.Global
-        public async Task<IActionResult> PublicationSubscribeFunc(
-            [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "publication/subscribe/")]
-            HttpRequest req,
-            FunctionContext context)
+        public async Task<IActionResult> RequestPendingSubscriptionFunc(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "publication/request-pending-subscription/")]
+            [FromBody] NewPendingSubscriptionRequest req,
+            FunctionContext context,
+            CancellationToken cancellationToken)
         {
-            _logger.LogInformation("{FunctionName} triggered", context.FunctionDefinition.Name);
+            logger.LogInformation("{FunctionName} triggered", context.FunctionDefinition.Name);
 
-            var subscriptionsTable = await _storageTableService.GetTable(NotifierSubscriptionsTableName);
-            var pendingSubscriptionsTable = await _storageTableService.GetTable(NotifierPendingSubscriptionsTableName);
+            var validationResult = await requestValidator.Validate(req, cancellationToken);
+            if (validationResult.IsLeft) // If validation failed
+            {
+                logger.LogError("{FunctionName} failed because the {req} did not contain a valid JSON object.", context.FunctionDefinition.Name, req);
+                return validationResult.Left;
```

Comment:
What sort of information is included in this result? Does it need to be "cleansed" or made more client-friendly before being returned (i.e. returned as a new `BadRequestResult` (or similar) with a custom message like the other methods)?